test/common/http/utility_test.cc

```diff
@@ -16,7 +16,9 @@ namespace Envoy {
 namespace Http {
 
 // Satisfy linker
-const uint64_t CodecOptions::NoCompression;
+const uint64_t Http2Settings::CodecOptions::DISABLE_DYNAMIC_HPACK_TABLE;
```

Comment:
Without this there is no storage actually assigned to store this static variable.  That may be OK: if you never require actual storage for the constant int the compiler doesn't complain.  But if the static const is passed by reference or it's address taken at some point in the code then it needs an actual storage location (if it's passed by reference someone might take the address later).  This means it must be _defined_ (it's _declared_ in the header) in a single CPP file.

Why does it matter here and not in the Envoy executable? In the Envoy executable case they're never passed by reference so the compiler lets you get away with not defining the variable.